### DISCUSSION TITLE
Pydantic Validation Error

### DISCUSSION THREAD

**Discussion by Tamunonengiyeofori** [pydantic/pydantic#4640] [Question]
When I try to create a new UserProfile and validate the response of the POST request with a validation schema for the UserProfile, I get a dialect validation error.
When I try to test a get request to return all the users and validate with a schema, the dialect validation still occurs. I don't know what I am doing wrong. Please I need help.

**Comment by samuelcolvin** [comment]
Sorry, @Tamunonengiyeofori you haven't given enough detail here for people to be able to help you.

**Reply by Tamunonengiyeofori** [reply]
# THE CODE FOR THE DBMODELS AND VALIDATION SCHEMA IS BELOW
## MODELS
from sqlalchemy import TIME, Column, String, Integer, Boolean, ForeignKey
from sqlalchemy.sql.sqltypes import TIMESTAMP
from sqlalchemy.sql.expression import text
from sqlalchemy_utils.types import ChoiceType
from sqlalchemy.orm import relationship
from .database import Base
from enum import IntFlag

class User(Base):
    __tablename__ = "user"
    id = Column(Integer, primary_key=True, nullable=False)
    email =  Column(String(80), nullable=False, unique=True)
    password = Column(String(100), nullable=False)   
    is_active = Column(Boolean, server_default="False", nullable=False)
    is_admin = Column(Boolean, server_default="False", nullable=False)
    created_at = Column(TIMESTAMP(timezone=True), nullable=False, server_default=text("now()"))
    updated_at = Column(TIMESTAMP(timezone=True), nullable=True, onupdate=text("now()"))
    profile = relationship("UserProfile", back_populates="owner", uselist=False)
    
    def __repr__(self):
        return f"{self.email}"
    
class UserProfile(Base):
    __tablename__ = "user_profile"
    id = Column(Integer, primary_key=True, nullable=False)
    name = Column(String(80), nullable=False)
    gender = Column(String(80), nullable=False)
    address = Column(String(250), nullable=False)
    mobile = Column(String(80), nullable=False)
    loans = relationship("Loan", back_populates="owner")
    created_at = Column(TIMESTAMP(timezone=True), nullable=False, server_default=text("now()"))
    updated_at = Column(TIMESTAMP(timezone=True), nullable=True, onupdate=text("now()"))
    owner_id = Column(Integer, ForeignKey("user.id", ondelete="CASCADE"), nullable=False)
    owner = relationship("User", back_populates="profile")
    
class LoanType(IntFlag):
    QUATERLY = 1
    BIANNUALL = 2
    ANNUAL = 3
    
class DurationType(IntFlag):
    THREE_MONTHS = 1
    SIX_MONTHS = 2
    TWELVE_MONTHS = 3
    

class PaymentStatus(IntFlag):
    PAID = 1
    DEBT = 2
    PAYING = 3
    
class Loan(Base):
    __tablename__ = "loan"
    id = Column(Integer, primary_key=True, nullable=False)
    amount = Column(Integer, default=0, nullable=False)
    type = Column(ChoiceType(LoanType, impl=Integer), nullable=False)
    payment_status = Column(ChoiceType(PaymentStatus, impl=Integer()), default=PaymentStatus.DEBT.value, nullable=False)
    # duration = Column(ChoiceType(DurationType, impl=Integer()))
    duration = Column(Integer, default=0, nullable=False)
    date_taken = Column(TIMESTAMP(timezone=True), server_default=text("now()"), nullable=False)
    description = Column(String(255), nullable=False)
    date_paid = Column(TIMESTAMP(timezone=True), nullable=True, onupdate=text("now()"))
    owner_id = Column(Integer, ForeignKey("user_profile.id", ondelete="CASCADE"))
    owner = relationship("UserProfile", back_populates="loans")
    
    def __repr__(self):
        return f"{self.id} {self.type}"

#VALIDATION SCHEMAS
from pydantic import BaseModel, EmailStr, constr
from typing import Optional, List
from datetime import datetime
# from app.models import LoanType
from .config import settings

class UserProfileBase(BaseModel):
    loans: Optional[List["LoanResponse"]] = None
    class Config:
        orm_mode = True
    
class UserProfileCreate(BaseModel):
    name: str
    gender: str
    address: str
    mobile: constr(min_length=11)

class UserProfileOut(UserProfileBase):
    id: int
    created_at: datetime
    updated_at: Optional[datetime]
    owner_id: int
    owner: "UserOut"
    
    

class UserBase(BaseModel):
    email : EmailStr
    is_active: bool = False
    is_admin: bool = False
    profile: Optional[UserProfileOut] = None
    
    class Config:
        orm_mode = True    

# class UserOut(BaseModel):
#     id: Optional[int]

class UserCreate(UserBase):
    password: str
    
class UserLogin(BaseModel):
    email: EmailStr
    password: str
    
class UserOut(UserBase):
    id : int
    is_admin: bool 
    created_at: datetime
    profile: Optional[UserProfileOut]

**Reply by Tamunonengiyeofori** [reply]
![image](https://user-images.githubusercontent.com/78642266/197275067-f63dd4c7-4be2-4d64-bf90-da88e6adcda4.png)
![image](https://user-images.githubusercontent.com/78642266/197275053-22bad3cd-1d39-4209-9864-25239ad4617d.png)
 The images showing the data for the request body used on postman

**Reply by Tamunonengiyeofori** [reply]
# This is the full traceback error 
INFO:     127.0.0.1:60699 - "GET /users/ HTTP/1.1" 500 Internal Server Error
ERROR:    Exception in ASGI application
Traceback (most recent call last):
  File "C:\Users\User\Desktop\Coding\LOAN_SERVICE_API\venv\lib\site-packages\uvicorn\protocols\http\h11_impl.py", line 366, in run_asgi
    result = await app(self.scope, self.receive, self.send)
  File "C:\Users\User\Desktop\Coding\LOAN_SERVICE_API\venv\lib\site-packages\uvicorn\middleware\proxy_headers.py", line 75, in __call__
    return await self.app(scope, receive, send)
  File "C:\Users\User\Desktop\Coding\LOAN_SERVICE_API\venv\lib\site-packages\fastapi\applications.py", line 269, in __call__
    await super().__call__(scope, receive, send)
  File "C:\Users\User\Desktop\Coding\LOAN_SERVICE_API\venv\lib\site-packages\starlette\applications.py", line 124, in __call__
    await self.middleware_stack(scope, receive, send)
  File "C:\Users\User\Desktop\Coding\LOAN_SERVICE_API\venv\lib\site-packages\starlette\middleware\errors.py", line 184, in __call__
    raise exc
  File "C:\Users\User\Desktop\Coding\LOAN_SERVICE_API\venv\lib\site-packages\starlette\middleware\errors.py", line 162, in __call__
    raise ValidationError(errors, field.type_)
pydantic.error_wrappers.ValidationError: 3 validation errors for UserOut
response -> 0 -> profile
  TypeEngine.dialect_impl() missing 1 required positional argument: 'dialect' (type=type_error)
response -> 1 -> profile
  TypeEngine.dialect_impl() missing 1 required positional argument: 'dialect' (type=type_error)
response -> 2 -> profile
  TypeEngine.dialect_impl() missing 1 required positional argument: 'dialect' (type=type_error

**Comment by samuelcolvin** [comment]
This is not how you get with your issue:
1. format your code
2. try to minimise the size of the example you give
3. make sure the issue is specific to pydantic, it looks like there are lots of other moving parts here.

I'm sorry I can't be more help, but it's very hard to give advice based on what you've put here.

**Reply by Tamunonengiyeofori** [reply]
from pydantic import BaseModel, EmailStr, constr
from typing import Optional, List
from datetime import datetime
# from app.models import LoanType
from .config import settings

class UserProfileBase(BaseModel):
    loans: Optional[List["LoanResponse"]] = None
    class Config:
        orm_mode = True


class UserProfileCreate(BaseModel):
    name: str
    gender: str
    address: str
    mobile: constr(min_length=11)


class UserProfileOut(UserProfileBase):
    id: int
    created_at: datetime
    updated_at: Optional[datetime]
    owner_id: int
    owner: "UserOut"


class UserBase(BaseModel):
    email : EmailStr
    is_active: bool = False
    is_admin: bool = False
    profile: Optional[UserProfileOut] = None
       
    class Config:
        orm_mode = True    

# class UserOut(BaseModel):
#     id: Optional[int]

class UserCreate(UserBase):
    password: str
    
class UserLogin(BaseModel):
    email: EmailStr
    password: str
    
class UserOut(UserBase):
    id : int
    is_admin: bool 
    created_at: datetime



class LoanResponse(BaseModel):
    id: int
    amount: int
    type: int
    payment_status: int
    duration: int
    date_taken: datetime
    description: str
    date_paid: Optional[datetime] = None
    owner_id: int
    owner: UserProfileOut
    
    class Config:
        orm_mode = True
        
class LoanCreate(BaseModel):
    amount: int
    type: int
    duration: int
    description: str
 
    
class Token(BaseModel):
    access_token: str
    token_type: str
    
class TokenData(BaseModel):
    id: Optional[str] = None


This schema is where the validation errors arises and it is based on pydantic

**Reply by Tamunonengiyeofori** [reply]
# error message 
response -> 1 -> profile
  TypeEngine.dialect_impl() missing 1 required positional argument: 'dialect' (type=type_error)
response -> 2 -> profile
  TypeEngine.dialect_impl() missing 1 required positional argument: 'dialect' (type=type_error)
